### PR TITLE
Add profile page for users and edit profile function

### PR DESCRIPTION
**Pull request by AmberWilkie**
PT story: https://www.pivotaltracker.com/story/show/130181999
and also this PT story, which obviously was a tiny change: https://www.pivotaltracker.com/story/show/130280231

Adds a profile page at /profile and fills it with information from the user's entry in the User table.
Also adds an edit profile route (the page already existed within the views Devise created) and updates Devise-basic edit user page to include other bits we want in the User model.

Does _not_ include sad path testing, as per our conversation.

### CODE REVIEW

File: app/controllers/profile_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class ProfileController < ApplicationController
+  def index
+  end
+end
```
Comment (tochman) [inline app/controllers/profile_controller.rb:4]: What is this controller used for?

File: app/views/devise/registrations/edit.html.erb [hunk 2]
```diff
@@ -12,7 +12,43 @@
...
   <% end %>
 
-  <div class="field">
+  <div class="pure-control-group">
+    <%= f.label :garden_website, 'Garden Website' %><br />
+    <%= f.text_field :garden_website %>
+  </div>
+
+  <div class="pure-control-group">
+    <%= f.label :blurb, 'Blurb' %><br />
```
Comment (tochman) [inline app/views/devise/registrations/edit.html.erb]: Not a blocker but you don't need to set the label if you are using 'Blurb' Rails will use `:blurb` and the label will be set to `Blurb` automatically. You only want to overwrite that if you actually want to use another label than the attribute name.

File: app/views/profile/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+<div>
+Name: <br>
+Garden Website: <%= link_to current_user.garden_website%><br>
+About Me: <%= current_user.blurb %><br>
+Garden Address:<br>
+<%= current_user.g_street %><br>
+<%= current_user.g_street2 %><br>
+<%= current_user.g_postcode %>, <%= current_user.g_city %>, <%= current_user.g_country %>
+
+</div>
```
Comment (tochman) [inline app/views/profile/index.html.erb:12]: Plz fix code formatting in this file.

File: app/controllers/profile_controller.rb (hunk 1 above)
Comment (AmberWilkie) [reply app/controllers/profile_controller.rb:4]: That controller displays the profile page each user has. We'll eventually need to make it more complex to allow only the current user to access certain parts of their own profile page.

Comment (AmberWilkie) [reply app/controllers/profile_controller.rb:4]: Does it work if I just reply here?

That controller displays the profile page each user has. We'll eventually
need to make it more complex to allow only the current user to access
certain parts of their own profile page.

_Amber Wilkie_
www.amberwilkie.com
0720.08.27.07 | Instagram http://www.instagram.com/amberwphoto

File: features/step_definitions/steps.rb [hunk 4]
```diff
@@ -15,5 +15,29 @@
...
+When(/^I click the "([^"]*)" link$/) do |link|
+  click_link(link)
+end
+
+Given(/^I am logged in$/) do
+  visit new_user_session_path
+  fill_in "Email", with: 'anna@random.com'
+  fill_in "Password", with: 'password'
+  click_button "Log in"
+end
```
Comment (tochman) [inline features/step_definitions/steps.rb:37]: there are Warden/Devise helpers you can (and out of performance perspective should) use to log in a user without going through the user interface. Not a blocker though.

**Comment by tochman** [comment]
I would like you to remove commented code.

File: app/views/devise/registrations/edit.html.erb (hunk 2 above)
Comment (AmberWilkie) [reply app/views/devise/registrations/edit.html.erb]: Understood.

File: features/step_definitions/steps.rb (hunk 4 above)
Comment (AmberWilkie) [reply features/step_definitions/steps.rb:37]: I saw that, but the docs say this is the "best" way to do this, though slower.

**Comment by AmberWilkie** [comment]
So I see what you're saying about commented code, but comments I leave in are there because I anticipate that they will be helpful in the future.

Does no one use commenting anymore? As I understand it, it's a very important thing in a lot of coding. In refactor we would of course remove anything that we didn't end up needing, but if I go down a road I know I will pick up later, I'd like to be able to leave myself breadcrumbs.

File: app/controllers/profile_controller.rb (hunk 1 above)
Comment (tochman) [reply app/controllers/profile_controller.rb:4]: Okay. Thx

File: features/step_definitions/steps.rb (hunk 4 above)
Comment (tochman) [reply features/step_definitions/steps.rb:37]: The docs re totally wrong ;-)

**Pull request merged**